src/frontend/src/catalog/system_catalog/rw_catalog/rw_sources.rs

```diff
@@ -12,25 +12,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::LazyLock;
+
 use risingwave_common::types::DataType;
 
 use crate::catalog::system_catalog::SystemCatalogColumnsDef;
 
 pub const RW_SOURCES_TABLE_NAME: &str = "rw_sources";
 
-pub const RW_SOURCES_COLUMNS: &[SystemCatalogColumnsDef<'_>] = &[
-    (DataType::Int32, "id"),
-    (DataType::Varchar, "name"),
-    (DataType::Int32, "schema_id"),
-    (DataType::Int32, "owner"),
-    (DataType::Varchar, "connector"),
-    // [col1, col2]
-    (DataType::Varchar, "columns"),
-    // [idx1, idx2]
-    (DataType::Varchar, "primary_keys"),
-    (DataType::Varchar, "row_format"),
-    (DataType::Boolean, "append_only"),
-    (DataType::Int32, "connection_id"),
-    (DataType::Varchar, "definition"),
-    (DataType::Varchar, "acl"),
-];
+pub static RW_SOURCES_COLUMNS: LazyLock<Vec<SystemCatalogColumnsDef<'_>>> = LazyLock::new(|| {
+    vec![
+        (DataType::Int32, "id"),
+        (DataType::Varchar, "name"),
+        (DataType::Int32, "schema_id"),
+        (DataType::Int32, "owner"),
+        (DataType::Varchar, "connector"),
+        // [col1, col2]
+        (DataType::List(Box::new(DataType::Varchar)), "columns"),
+        // [idx1, idx2]
+        (DataType::List(Box::new(DataType::Int32)), "primary_keys"),
```

Comment:
it does not make sense that source has a primary key. I think current field in proto is just because of historical reason. So I think we should remove it in the system table.

Reply:
🤔 Indeed, it's only required by table source. Let's only expose stream source here and remove it.